Documentation is unclear; I cannot get the crate to work.

I'll preface this by saying that *I* could be doing something wrong, but I've approached this just like any other crate; include it, build it, and then try to get an example working.

This is the point I've gotten stuck at, loading a font: [line 65 of example](https://github.com/alexheretic/glyph-brush/blob/master/glyph-brush/examples/opengl.rs#L65).

I've written the following:
```rust
let typeface = FontRef::try_from_slice(include_bytes!(r#"C:\Users\Kuba\Desktop\FiraCode-Regular.ttf"#)).expect("a font");
let mut glyph_brush = GlyphBrushBuilder::using_font(typeface).build();
```
And I get the following error:
```
error[E0282]: type annotations needed for `GlyphBrush<V, X, glyph_brush::ab_glyph::FontRef<'_>>`
   --> src\main.rs:236:64
    |
236 |     let mut glyph_brush = GlyphBrushBuilder::using_font(typeface).build();
    |         ---------------                                           ^^^^^ cannot infer type for type parameter `V` declared on the associated function `build`
    |         |
    |         consider giving `glyph_brush` the explicit type `GlyphBrush<V, X, glyph_brush::ab_glyph::FontRef<'_>>`, where the type parameter `V` is specified
```
That example doesn't include any type annotations at all. I have looked through the docs, and for the life of me I cannot figure out the type annotation required. There are no examples aside from this one.

I've managed to get something to compile by using a `FontArc` instead of a `FontRef`, and type annotating `GlyphBrush<GlyphVertex>` but that isn't what the example is showing. What's going wrong?

--- a/glyph-brush/CHANGELOG.md
+++ b/glyph-brush/CHANGELOG.md
@@ -1,3 +1,6 @@
+# Unreleased
+* Add documentation of `GlyphBrush` generic types & workarounds when `.build()` type inference fails.
+
 # 0.7.2
 * Add `GlyphBrushBuilder::multithread` to allow setting the (default on) draw-cache multithreading.
 

--- a/glyph-brush/src/glyph_brush.rs
+++ b/glyph-brush/src/glyph_brush.rs
@@ -18,22 +18,27 @@ type SectionHash = u64;
 /// Object allowing glyph drawing, containing cache state. Manages glyph positioning cacheing,
 /// glyph draw caching & efficient GPU texture cache updating.
 ///
-/// Build using a [`GlyphBrushBuilder`](struct.GlyphBrushBuilder.html).
+/// Build using a [`GlyphBrushBuilder`].
 ///
-/// Also see [`GlyphCruncher`](trait.GlyphCruncher.html) trait which providers extra functionality,
-/// such as [`glyph_bounds`](trait.GlyphCruncher.html#method.glyph_bounds).
+/// Also see [`GlyphCruncher`] trait which providers extra functionality,
+/// such as [`GlyphCruncher::glyph_bounds`].
+///
+/// # Generic types
+/// * **`V`** A single glyph's vertex data type that matches, or is inferred by, the `to_vertex`
+///   function given to the [`GlyphBrush::process_queued`] call.
+/// * **`X`** Extra non-layout data for use in vertex generation. _Default [`Extra`]_.
+/// * **`F`** _ab-glyph_ font type. _Default [`FontArc`]_.
+/// * **`H`** Section hasher used for cache matching. _Default [`DefaultSectionHasher`]_.
 ///
 /// # Caching behaviour
-///
-/// Calls to [`GlyphBrush::queue`](#method.queue),
-/// [`GlyphBrush::glyph_bounds`](#method.glyph_bounds), [`GlyphBrush::glyphs`](#method.glyphs)
+/// Calls to [`GlyphBrush::queue`], [`GlyphBrush::glyph_bounds`], [`GlyphBrush::glyphs`]
 /// calculate the positioned glyphs for a section.
 /// This is cached so future calls to any of the methods for the same section are much
-/// cheaper. In the case of [`GlyphBrush::queue`](#method.queue) the calculations will also be
+/// cheaper. In the case of [`GlyphBrush::queue`] the calculations will also be
 /// used for actual drawing.
 ///
 /// The cache for a section will be **cleared** after a
-/// [`GlyphBrush::process_queued`](#method.process_queued) call when that section has not been used
+/// [`GlyphBrush::process_queued`] call when that section has not been used
 /// since the previous call.
 ///
 /// # Texture caching behaviour
@@ -42,9 +47,7 @@ type SectionHash = u64;
 /// This is required for high quality text as a glyph's positioning is always exactly aligned
 /// to it's draw positioning.
 ///
-/// This behaviour can be adjusted with
-/// [`GlyphBrushBuilder::draw_cache_position_tolerance`]
-/// (struct.GlyphBrushBuilder.html#method.draw_cache_position_tolerance).
+/// This behaviour can be adjusted with [`GlyphBrushBuilder::draw_cache_position_tolerance`].
 pub struct GlyphBrush<V, X = Extra, F = FontArc, H = DefaultSectionHasher> {
     fonts: Vec<F>,
     texture_cache: DrawCache,

--- a/glyph-brush/src/glyph_brush.rs
+++ b/glyph-brush/src/glyph_brush.rs
@@ -27,8 +27,9 @@ type SectionHash = u64;
 /// * **`V`** A single glyph's vertex data type that matches, or is inferred by, the `to_vertex`
 ///   function given to the [`GlyphBrush::process_queued`] call.
 /// * **`X`** Extra non-layout data for use in vertex generation. _Default [`Extra`]_.
-/// * **`F`** _ab-glyph_ font type. _Default [`FontArc`]_.
-/// * **`H`** Section hasher used for cache matching. _Default [`DefaultSectionHasher`]_.
+/// * **`F`** _ab-glyph_ font type. Generally inferred by builder usage. _Default [`FontArc`]_.
+/// * **`H`** Section hasher used for cache matching. See [`GlyphBrushBuilder::section_hasher`].
+///   _Default [`DefaultSectionHasher`]_.
 ///
 /// # Caching behaviour
 /// Calls to [`GlyphBrush::queue`], [`GlyphBrush::glyph_bounds`], [`GlyphBrush::glyphs`]

--- a/glyph-brush/src/glyph_brush/builder.rs
+++ b/glyph-brush/src/glyph_brush/builder.rs
@@ -3,10 +3,9 @@ use glyph_brush_draw_cache::*;
 use glyph_brush_layout::ab_glyph::*;
 use std::hash::BuildHasher;
 
-/// Builder for a [`GlyphBrush`](struct.GlyphBrush.html).
+/// Builder for a [`GlyphBrush`].
 ///
 /// # Example
-///
 /// ```
 /// use glyph_brush::{ab_glyph::FontArc, GlyphBrush, GlyphBrushBuilder};
 /// # type Vertex = ();
@@ -183,7 +182,7 @@ impl<F: Font, H: BuildHasher> GlyphBrushBuilder<F, H> {
         self
     }
 
-    /// Sets the section hasher. `GlyphBrush` cannot handle absolute section hash collisions
+    /// Sets the section hasher. [`GlyphBrush`] cannot handle absolute section hash collisions
     /// so use a good hash algorithm.
     ///
     /// This hasher is used to distinguish sections, rather than for hashmap internal use.
@@ -210,7 +209,7 @@ impl<F: Font, H: BuildHasher> GlyphBrushBuilder<F, H> {
         }
     }
 
-    /// Builds a `GlyphBrush`.
+    /// Builds a [`GlyphBrush`].
     ///
     /// If type inference fails try declaring the types `V` & `X`.
     /// See [`GlyphBrush` generic types](struct.GlyphBrush.html#generic-types).
@@ -246,7 +245,7 @@ impl<F: Font, H: BuildHasher> GlyphBrushBuilder<F, H> {
         }
     }
 
-    /// Rebuilds an existing `GlyphBrush` with this builder's properties. This will clear all
+    /// Rebuilds an existing [`GlyphBrush`] with this builder's properties. This will clear all
     /// caches and queues.
     ///
     /// # Example

--- a/glyph-brush/src/glyph_brush/builder.rs
+++ b/glyph-brush/src/glyph_brush/builder.rs
@@ -210,7 +210,18 @@ impl<F: Font, H: BuildHasher> GlyphBrushBuilder<F, H> {
         }
     }
 
-    /// Builds a `GlyphBrush` using the input gfx factory
+    /// Builds a `GlyphBrush`.
+    ///
+    /// If type inference fails try declaring the types `V` & `X`.
+    /// See [`GlyphBrush` generic types](struct.GlyphBrush.html#generic-types).
+    /// ```
+    /// # use glyph_brush::{ab_glyph::*, GlyphBrushBuilder};
+    /// # let some_font = FontArc::try_from_slice(include_bytes!("../../../fonts/DejaVuSans.ttf")).unwrap();
+    /// # type SomeOtherBuildHasher = glyph_brush::DefaultSectionHasher;
+    /// # type Vertex = ();
+    /// let glyph_brush = GlyphBrushBuilder::using_font(some_font)
+    ///     .build::<Vertex, glyph_brush::Extra>();
+    /// ```
     pub fn build<V, X>(self) -> GlyphBrush<V, X, F, H> {
         GlyphBrush {
             fonts: self.font_data,

